WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListActivity.java

```diff
@@ -0,0 +1,288 @@
+        }
+
+        if (savedInstanceState == null) {
+            PublicizeTable.createTables(WordPress.wpDB.getDatabase());
+            mSiteId = getIntent().getIntExtra(PublicizeConstants.ARG_SITE_ID, 0);
+            showListFragment(mSiteId);
+            PublicizeUpdateService.updateConnectionsForSite(this, mSiteId);
+        } else {
+            mSiteId = savedInstanceState.getInt(PublicizeConstants.ARG_SITE_ID);
+        }
+
+        getFragmentManager().addOnBackStackChangedListener(new FragmentManager.OnBackStackChangedListener() {
+            @Override
+            public void onBackStackChanged() {
+                // clear the nav icon and subtitle used for the webView fragment
+                if (getFragmentManager().getBackStackEntryCount() < 2) {
+                    mToolbar.setNavigationIcon(R.drawable.ic_arrow_back_white_24dp);
+                    mToolbar.setSubtitle(null);
+                }
+            }
+        });
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putInt(PublicizeConstants.ARG_SITE_ID, mSiteId);
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        EventBus.getDefault().unregister(this);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        EventBus.getDefault().register(this);
+    }
+
+    private void showListFragment(int siteId) {
+        if (isFinishing()) return;
+
+        String tag = getString(R.string.fragment_tag_publicize_list);
+        Fragment fragment = PublicizeListFragment.newInstance(siteId);
+        getFragmentManager()
+                .beginTransaction()
+                .replace(R.id.fragment_container, fragment, tag)
+                .commit();
+    }
+
+    private PublicizeListFragment getListFragment() {
+        String tag = getString(R.string.fragment_tag_publicize_list);
+        Fragment fragment = getFragmentManager().findFragmentByTag(tag);
+        if (fragment != null) {
+            return (PublicizeListFragment) fragment;
+        } else {
+            return null;
+        }
+    }
+
+    private void reloadListFragment() {
+        PublicizeListFragment listFragment = getListFragment();
+        if (listFragment != null) {
+            listFragment.reload();
+        }
+    }
+
+    /*
+     * close all but the first (list) fragment
+     */
+    private void returnToListFragment() {
+        if (getFragmentManager().getBackStackEntryCount() == 0) return;
+
+        String tag = getString(R.string.fragment_tag_publicize_detail);
+        getFragmentManager().popBackStack(tag, FragmentManager.POP_BACK_STACK_INCLUSIVE);
+    }
+
+    private void showDetailFragment(int siteId,
+                                    PublicizeService service,
+                                    PublicizeConnection connection) {
+        if (isFinishing()) return;
+
```

Comment:
Not necessary, but it feels weird when you hit back from a detail view and have the same title in the action bar: https://cloudup.com/cABp1g5qoGg